src/ucs/config/parser.c

```diff
@@ -426,24 +551,15 @@ int ucs_config_sscanf_memunits(const char *buf, void *dest, const void *arg)
     if (num_fields == 1) {
         bytes = 1;
     } else if (num_fields == 2 || num_fields == 3) {
-        if (!strcasecmp(units, "b")) {
-            bytes = 1;
-        } else if (!strcasecmp(units, "kb") || !strcasecmp(units, "k")) {
-            bytes = UCS_KBYTE;
-        } else if (!strcasecmp(units, "mb") || !strcasecmp(units, "m")) {
-            bytes = UCS_MBYTE;
-        } else if (!strcasecmp(units, "gb") || !strcasecmp(units, "g")) {
-            bytes = UCS_GBYTE;
-        } else if (!strcasecmp(units, "tb") || !strcasecmp(units, "t")) {
-            bytes = UCS_TBYTE;
-        } else {
+        bytes = ucs_config_parser_quantity_prefix_value(toupper(units[0]));
+        if (!bytes || ((num_fields == 3) && tolower(units[1]) != 'b')) {
             return 0;
         }
     } else {
         return 0;
     }
 
-    *(size_t*)dest = value * bytes;
+    *(size_t*)dest = (size_t)(value * bytes);
```

Comment:
why need to cast to size_t?

Reply:
double->size_t automatic replace issue :)
fixed